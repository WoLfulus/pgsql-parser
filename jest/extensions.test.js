const parser = require('../src');
import { cleanTree, cleanLines } from './utils';
const dosql = require('fs').readFileSync(__dirname + '/fixtures/extensions.sql').toString();

describe('extensions', () => {
  it('parse and deparse', () => {
    const tree = parser.parse(dosql);
    expect(tree).toMatchSnapshot();
    const sql = parser.deparse(tree.query);
    expect(cleanLines(sql)).toMatchSnapshot();
    expect(cleanTree(parser.parse(sql))).toEqual(cleanTree(tree));
  });
});
