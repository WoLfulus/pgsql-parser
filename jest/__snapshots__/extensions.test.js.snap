// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extensions parse and deparse 1`] = `
Object {
  "query": Array [
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "CreateExtensionStmt": Object {
            "extname": "awesome-roles",
            "options": Array [
              Object {
                "DefElem": Object {
                  "arg": Object {
                    "Integer": Object {
                      "ival": 1,
                    },
                  },
                  "defaction": 0,
                  "defname": "cascade",
                  "location": 33,
                },
              },
            ],
          },
        },
        "stmt_len": 40,
      },
    },
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "CreateExtensionStmt": Object {
            "extname": "awesome-roles",
            "if_not_exists": true,
          },
        },
        "stmt_len": 47,
        "stmt_location": 41,
      },
    },
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "CreateExtensionStmt": Object {
            "extname": "awesome-roles",
            "if_not_exists": true,
            "options": Array [
              Object {
                "DefElem": Object {
                  "arg": Object {
                    "Integer": Object {
                      "ival": 1,
                    },
                  },
                  "defaction": 0,
                  "defname": "cascade",
                  "location": 137,
                },
              },
            ],
          },
        },
        "stmt_len": 55,
        "stmt_location": 89,
      },
    },
  ],
  "stderr": "",
}
`;

exports[`extensions parse and deparse 2`] = `
"CREATE EXTENSION \\"awesome-roles\\" CASCADE;
CREATE EXTENSION IF NOT EXISTS \\"awesome-roles\\";
CREATE EXTENSION IF NOT EXISTS \\"awesome-roles\\" CASCADE;"
`;
